pkg/driver/controller.go

```diff
@@ -206,28 +206,66 @@ func (d *ControllerService) CreateVolume(ctx context.Context, req *csi.CreateVol
 			ext4ClusterSize = value
 		default:
 			if strings.HasPrefix(key, TagKeyPrefix) {
-				scTags = append(scTags, value)
+				tagsToEvaluate = append(tagsToEvaluate, value)
 			} else {
 				return nil, status.Errorf(codes.InvalidArgument, "Invalid parameter key %s for CreateVolume", key)
 			}
 		}
 	}
 
-	modifyOptions, err := parseModifyVolumeParameters(req.GetMutableParameters())
-	if err != nil {
-		return nil, status.Errorf(codes.InvalidArgument, "Invalid mutable parameter: %v", err)
-	}
+	mutableParameters := req.GetMutableParameters()
 
 	// "Values specified in mutable_parameters MUST take precedence over the values from parameters."
 	// https://github.com/container-storage-interface/spec/blob/master/spec.md#createvolume
-	if modifyOptions.modifyDiskOptions.VolumeType != "" {
-		volumeType = modifyOptions.modifyDiskOptions.VolumeType
+	// Iterating mutableParameters separately to not incorrectly set default values on creation in case of parameters not having these set.
+	for key, value := range mutableParameters {
+		switch key {
+		case IopsKey:
+			vacIops, err := strconv.ParseInt(value, 10, 32)
+			if err != nil {
+				return nil, status.Errorf(codes.InvalidArgument, "Could not parse IOPS: %q", value)
+			}
+			iops = int32(vacIops)
+		case ThroughputKey:
+			vacThroughput, err := strconv.ParseInt(value, 10, 32)
+			if err != nil {
+				return nil, status.Errorf(codes.InvalidArgument, "Could not parse throughput: %q", value)
+			}
+			throughput = int32(vacThroughput)
+		case DeprecatedModificationKeyVolumeType:
+			if _, ok := mutableParameters[ModificationKeyVolumeType]; ok {
+				klog.Infof("Ignoring deprecated key `volumeType` because preferred key `type` is present")
+				continue
+			}
+			klog.InfoS("Key `volumeType` is deprecated, please use `type` instead")
+			volumeType = value
+		case VolumeTypeKey:
+			volumeType = value
+		default:
+			switch {
+			case strings.HasPrefix(key, ModificationAddTag):
+				tagsToEvaluate = append(tagsToEvaluate, value)
```

Comment:
How does this handle a tag value being specified in both the SC and VAC parameters?